viewer/src/main/webapp/viewer-html/components/EditBulk.js

```diff
@@ -0,0 +1,396 @@
+                return;
+            }
+
+            var me = this;
+
+
+            var features = this.selectedFeatures.getSelectedFeaturesWithChangesApplied();
+
+            for (var featureIndex = 0; featureIndex < features.length; featureIndex++) {
+                var feature = features[featureIndex];
+                me.editingLayer = this.config.viewerController.getLayer(this.layerSelector.getValue());
+                var applayerId = me.editingLayer.getId();
+                this.lastUsedValues [applayerId] = feature;
+
+                if (this.geometryEditable) {
+                    if (this.vectorLayer.getActiveFeature()) {
+                        var wkt = this.vectorLayer.getActiveFeature().config.wktgeom;
+                        feature[this.appLayer.geometryAttribute] = wkt;
+                    }
+                    if (!feature[this.appLayer.geometryAttribute]) {
+                        return;
+                    }
+                }
+                if (this.mode === "edit") {
+                    feature.__fid = features[featureIndex].__fid;
+                }
+                if (this.mode === "copy") {
+                    this.currentFID = null;
+                    delete feature.__fid;
+                }
+                try {
+                    feature = this.changeFeatureBeforeSave(feature);
+                } catch (e) {
+                    me.failed(e);
+                    return;
+                }
+                var ef = this.getEditFeature();
+                ef.edit(
+                    me.editingLayer,
+                    feature,
+                    function (fid) {
+                        // me.saveSucces(fid);
+                        // me.config.viewerController.fireEvent(viewer.viewercontroller.controller.Event.ON_EDIT_SUCCESS, me.editingLayer, feature);
+                        console.debug('success!');
+                    }, function (error) {
+                        // me.failed(error);
+                        console.debug('failed :(');
+                    });
+            }
+
+            var msg = i18next.t('viewer_components_edit_34');
+            Ext.Msg.alert("Gelukt", msg);
+            this.cancel();
+        },
+        onFormFieldChange: function(me, input, newValue, oldValue) {
+            var attributeName = input.name;
+            me.selectedFeatures.changeValue(attributeName, newValue);
+            me.makeFormFieldScalar(attributeName);
+        },
+        getListOfAttributeNamesThatAreMappedToFormFields: function () {
+            var me = this;
+            var mapped = [];
+            me.appLayer.attributes.forEach(function(attribute) {
+                var input = me.inputContainer.getForm().findField(attribute.name);
+                if (input !== null) {
+                    mapped.push(attribute.name);
+                }
+            });
+            return mapped;
+        }
+    });
+
+    function SelectedFeatures(mapComponent, attributeNames) {
```

Comment:
I'm more in favor of using one style of class creation, so please make it a Ext class, in its own file